doc/developer-guides/coding_guidelines.rst

```diff
@@ -695,6 +724,294 @@ Compliant example::
        }
 
 
+FN-11: The return type of a function shall be consistent
+========================================================
+
+The return type of a function shall be the same between its declaration and
+definition.
+
+Compliant example::
+
+    /* In `showcase.h`: */
+    uint32_t func_showcase(uint32_t param);
+    
+    /* In `showcase.c`: */
+    #include "showcase.h"
+    
+    uint32_t func_showcase(uint32_t param)
+    {
+        return param;
+    }
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       /* In `showcase.h`: */
+       uint64_t func_showcase(uint64_t param);
+       
+       /* In `showcase.c`: */
+       #include "showcase.h"
+       
+       uint32_t func_showcase(uint32_t param)
+       {
+               return param;
+       }
+
+
+FN-12: Banned functions shall not be used
+=========================================
+
+The following cases shall be covered:
```

Comment:
This would read better as:
```
a) These dynamic memory allocation functions shall not be used:
    ``calloc``, ``malloc``, ``realloc``, and ``free``.  Dynamic memory
   allocation shall be replaced with static memory allocation.
b) The functions ``va_arg``, ``va_start``, and ``va_end`` shall
   only be used within variadic functions (functions taking a
   variable number of parameters) such as ``printf``.
```